src/lib.rs

```diff
@@ -1,66 +1,248 @@
-#![allow(non_snake_case)]
-
 //!
 //! Parsing of ABNF Core Rules
 //!
-//! See https://tools.ietf.org/html/rfc5234#appendix-B.1
+//! See <https://tools.ietf.org/html/rfc5234#appendix-B.1>
 //!
 
 pub mod complete;
 pub mod streaming;
 
-pub fn is_alpha(c: char) -> bool {
-    c.is_ascii_alphabetic()
+use nom::AsChar;
+
+/// A-Z / a-z
+///
+/// ALPHA = %x41-5A / %x61-7A
+pub fn is_alpha(c: impl AsChar) -> bool {
+    matches!(c.as_char(), '\x41'..='\x5A' | '\x61'..='\x7A')
+}
```

Comment:
I used `matches` and hexa escape code as much as possible to avoid mistake, a review of the code can look the ABNF core and quickly conclude it's correct, without worry about rust character compatibility.